src/GitVersionCore/Configuration/ConfigurationProvider.cs

```diff
@@ -31,15 +31,15 @@ public static void ApplyDefaultsTo(Config config)
             var configBranches = config.Branches.ToList();
 
             ApplyBranchDefaults(config, GetOrCreateBranchDefaults(config, "master"), defaultTag: string.Empty, defaultPreventIncrement: true);
-            ApplyBranchDefaults(config, GetOrCreateBranchDefaults(config, "release[/-]"), defaultTag: "beta", defaultPreventIncrement: true);
```

Comment:
Add another parameter to this which is `migrate:` and then it takes an array. You can then add `migrate: new[] { "release[/-]" }` and ApplyBranchDefaults should rename the configs to be migrated, then apply the defaults.

Just one idea on how to solve.